plugin-lifecycle/src/main/java/com/mapbox/maps/plugin/lifecycle/MapboxLifecyclePluginImpl.kt

```diff
@@ -10,58 +10,53 @@ import android.content.ComponentCallbacks2.TRIM_MEMORY_RUNNING_MODERATE
 import android.content.ComponentCallbacks2.TRIM_MEMORY_UI_HIDDEN
 import android.content.res.Configuration
 import android.view.View
-import androidx.lifecycle.Lifecycle
-import androidx.lifecycle.LifecycleObserver
-import androidx.lifecycle.OnLifecycleEvent
-import androidx.lifecycle.ViewTreeLifecycleOwner
+import androidx.lifecycle.*
 import com.mapbox.maps.MapboxLifecycleObserver
-import com.mapbox.maps.logE
 import com.mapbox.maps.logW
 import com.mapbox.maps.plugin.Plugin.Companion.MAPBOX_LIFECYCLE_PLUGIN_ID
 import com.mapbox.maps.plugin.delegates.MapPluginProviderDelegate
 
 /**
  * Concrete implementation of MapboxLifecyclePlugin.
  */
-class MapboxLifecyclePluginImpl : MapboxLifecyclePlugin {
+class MapboxLifecyclePluginImpl: MapboxLifecyclePlugin {
+  private lateinit var viewLifecycleRegistry: ViewLifecycleRegistry
+
   /**
    * Register a MapboxLifecycleObserver to observe life cycle events from LifecycleOwner
    *
    * @param mapView the instance of mapView, will get the LifecycleOwner from mapview's parent
    * @param observer the observer that listen to the life cycle events
    */
   override fun registerLifecycleObserver(mapView: View, observer: MapboxLifecycleObserver) {
-    val lifecycleOwner = ViewTreeLifecycleOwner.get(mapView)
-    if (lifecycleOwner == null) {
-      logE(
-        TAG,
-        """Can't get lifecycleOwner for mapview,
-          please make sure the host Activity is AppCompatActivity and the version of appcompat is 1.3.0+.
-          If the host Activity is not AppCompatActivity,
-          you need manually invoke the corresponding lifecycle methods in onStart/onStop/onDestroy/onLowMemory methods of the host Activity"""
+    if (!this::viewLifecycleRegistry.isInitialized) {
```

Comment:
Suppose `lateinit` is not needed and `val viewLifecycleRegistry: ViewLifecycleRegistry`  can be declared here now?